### PR TITLE
perf-counter: support P999 for percentile type

### PR DESCRIPTION
**Pull request by neverchanje**
This PR we add support for P999 latency metrics. So now for every COUNTER_TYPE_NUMBER_PERCENTILES, e.g
```
zion*profiler*RPC_RRDB_RRDB_PUT.latency.server
```
we record its P999 value during `take_snapshot`, giving its name as
```
zion*profiler*RPC_RRDB_RRDB_PUT.latency.server.p999
```

### CODE REVIEW

File: src/core/perf_counter/perf_counters.cpp [hunk 1]
```diff
@@ -222,17 +214,26 @@ void perf_counters::take_snapshot()
...
+            cs.type = c->type();
         }
         cs.updated_recently = true;
         if (c->type() != COUNTER_TYPE_NUMBER_PERCENTILES) {
             cs.value = c->get_value();
         } else {
             cs.value = c->get_percentile(COUNTER_PERCENTILE_99);
+
+            // take P999 metrics into account as well.
+            std::string name_p999 = std::string(c->full_name()) + ".999";
```
Comment (shengofsun) [inline src/core/perf_counter/perf_counters.cpp]: 叫p999清晰一些? "999"整的名字怪怪的

Comment (neverchanje) [reply src/core/perf_counter/perf_counters.cpp]: fixed

File: src/core/perf_counter/perf_counters.cpp [hunk 2]
```diff
@@ -222,17 +214,26 @@ void perf_counters::take_snapshot()
...
+            cs.type = c->type();
         }
         cs.updated_recently = true;
         if (c->type() != COUNTER_TYPE_NUMBER_PERCENTILES) {
             cs.value = c->get_value();
         } else {
             cs.value = c->get_percentile(COUNTER_PERCENTILE_99);
+
+            // take P999 metrics into account as well.
+            std::string name_p999 = std::string(c->full_name()) + ".p999";
```
Comment (qinzuoyan) [inline src/core/perf_counter/perf_counters.cpp]: 你在pr的说明中给的样例是：
```
zion*profiler*RPC_RRDB_RRDB_PUT.latency.server.999
```
这里却是`p999`。到底是哪个呢？

Comment (neverchanje) [reply src/core/perf_counter/perf_counters.cpp]: p999,按照伟杰的建议改了

File: src/core/tests/perf_counters_test.cpp [hunk 3]
```diff
@@ -40,7 +40,7 @@
 
 using namespace ::dsn;
 
-TEST(perf_counters, counter_create_remove)
+TEST(perf_counters_test, counter_create_remove)
```
Comment (vagetablechicken) [inline src/core/tests/perf_counters_test.cpp]: 这是都要改成尾缀_test的风格么？

Comment (neverchanje) [reply src/core/tests/perf_counters_test.cpp]: 重名的改，目前没有严格要求，按照个人习惯，google 的代码风格是 test suite name 加后缀 "Test"

**Pull request merged**